sdk/python/kfp/components/component_decorator.py

```diff
@@ -91,6 +91,10 @@ def pipeline():
     Returns:
         A component task factory that can be used in pipeline definitions.
     """
+    if output_component_file is not None:
+        raise Exception("output_component_file is not supported yet in v2 early"
```

Comment:
nit: `NotImplementedError` might be a better choice here than `Exception`.

No need to create a PR for this, but if we ever touch this file again before we implement it, we may consider updating it.

/cc @ji-yaqi @connor-mccarthy